### ISSUE TITLE
Consider not instrumenting a11y strings

### ISSUE THREAD

**Issue by jbphet** [phetsims/chipper#1352]
A11y string, i.e. those used for description, are currently showing up in Studio, meaning that they are instrumented.  Here is a screenshot from Greenhouse Effect:

![image](https://user-images.githubusercontent.com/3082831/200929138-4e709b76-d228-4f95-acb4-c980a745caa7.png)

This seems a bit odd, so I asked about it on Slack.  The dialog is below, but the summary is that we should consider not instrumenting these strings.

> @jbphet
>  [Today at 11:58 AM]
> I have a question about the intersection between a11y strings and phet-io.  While finishing up the conversion of Greenhouse Effect to use all StringProperties, I noticed that the a11y StringProperties are visible in Studio.  My understanding is that we are currently a long way from supporting dynamic locales and strings for description, so I'm wondering what the plans are for these StringProperties.  Should we consider filtering them out and not showing them in Studio?  Are we showing them because we do intend to support dynamic a11y strings fairly soon?  PS: I'm happy to log a github issue on this if y'all think I should.

> 4 replies

> @pixelzoom 
> Good point. All StringProperties are currently “featured”, including a11y strings. And if we don’t exclude instrumentation of a11y strings, we’ll need migration rules if the approach to a11y strings changes - which seems probable.  Sounds worthy of a GitHub issue.  (edited) 

> @pixelzoom 
> So I wouldn’t hide them in Studio — I just wouldn’t instrument a11y strings for now. Hiding them in studio doesn’t remove them from the PhET-iO API.  (edited) 

> @zepumph 
> I thought we were not instrumenting a11y strings. . . Will you make an issue please?

**Comment by jbphet** [comment]
@arouinfar wanted to make sure that we are talking about description strings here, and not *all* strings that relate to accessibility.  The rule of thumb should be that if a string shows up in the translation utility, it should show up in Studio.

**Comment by samreid** [comment]
What is the rule for whether a string appears in the translation utility?

**Comment by zepumph** [comment]
I don't really understand how @jbphet is doing this currently for greenhouse (which is what was suggested during PhET-iO meeting today), but can we do it generally and opt out of everying under a11y?


```diff
Index: js/getStringModule.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/getStringModule.ts b/js/getStringModule.ts
--- a/js/getStringModule.ts	(revision ef106e212501bbf87f137431b4089e8cd60c17c0)
+++ b/js/getStringModule.ts	(date 1668122165250)
@@ -189,6 +189,10 @@
         tandem = tandem.createTandem( tandemName );
       }
 
+      if ( tandem.name.includes( 'a11y' ) ) {
+        tandem = Tandem.OPT_OUT;
+      }
+
       const localizedString = new LocalizedString(
         phet.chipper.mapString( phet.chipper.strings[ FALLBACK_LOCALE ][ stringKey ] ),
         tandem,

**Comment by jbphet** [comment]
@samreid asked:

> What is the rule for whether a string appears in the translation utility?

A string appears in the translation utility if it is used in the built version of the sim and is not a sub-key of the `a11y` key.

@zepumph said:

> I don't really understand how @jbphet is doing this currently for greenhouse (which is what was suggested during PhET-iO meeting today), but can we do it generally and opt out of everything under a11y?

I'm not doing anything particular for Greenhouse Effect, I'm just observing that the strings are showing up in Studio, so yes, I think we should generally opt out of instrumenting all a11y strings.  The patch above looks reasonable and would probably work out fine, but I'm a little worried that the test isn't specific enough.  Can we test for `<repo-name>.a11y.` instead of any of `.a11y.`?  The latter seems like it could potentially bite us in a hard-to-debug way.

Back to you guys for the next steps.  I'm happy to implement it if you like, and I'm not sure if you passed it back to me for discussion or to just do it, so I thought I'd double check.

**Comment by samreid** [comment]
I committed a proposed fix, @jbphet can you please review?

**Comment by jbphet** [comment]
I just tested out the change using the master version of greenhouse-effect, and while all the sim-specific a11y strings appear to be gone, the common-code ones are still there.  Please have a look at the screenshot below.  Looking at the code change, I can see why this is the case.  @samreid - I'm thinking that *all* a11y strings should be removed.  Do you agree?

![image](https://user-images.githubusercontent.com/3082831/203166165-7c495efc-c473-4861-bf21-f3beedc2a300.png)

**Comment by zepumph** [comment]
Sorry I let this slide, but @jbphet's comment is exactly why I recommend leaving it general, and NOT including the sim name. Something like `.a11y.` will cover all repos.

**Comment by samreid** [comment]
I removed the rest of the a11y strings based on @zepumph recommendation, @jbphet can you please review?

**Comment by jbphet** [comment]
The commits look good, and I checked the behavior using Greenhouse Effect in Studio and didn't see any a11y strings.  Thanks.

I now better understand the need for generality, though I'm still a bit concerned that there could be a legitimate string with ".a11y." in it that *isn't* part of the subset of accessibility strings.  I suppose that's pretty unlikely, so I'm cool with not worrying about it until or unless it becomes a problem.  Closing.
